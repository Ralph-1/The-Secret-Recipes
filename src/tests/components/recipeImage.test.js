import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Image from '../../components/RecipeImage';
import renderer from 'react-test-renderer';

Enzyme.configure({ adapter: new Adapter() });
const setup = () => {
  const component = shallow(<Image src="jest" name="testing" />);
  return component;
};

describe('Header Component', () => {
  it('should match snapshot', () => {
    const tree = renderer.create(<Image src="jest" name="testing" />)

    expect(tree).toMatchSnapshot()
  });

  let component;
  beforeEach(() => {
    component = setup();
  });

  it('should display one image', () => {
    expect(component.find('img').hasClass('recipeImage')).toBe(true);
  });

  it('should should not display any image', () => {
    expect(component.find('img').hasClass('rcp-image')).toBe(false);
  });

  it('should display one header', () => {
    expect(component.find('h3').length).toBe(1);
  });

  it('should not display header ', () => {
    expect(component.find('h1').length).toBe(0);
  });
});
