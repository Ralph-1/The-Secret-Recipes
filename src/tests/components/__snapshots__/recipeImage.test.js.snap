// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component should match snapshot 1`] = `
<div
  className="w-50"
>
  <img
    alt="testing"
    className="recipeImage h4 bold"
    name="testing"
    src="jest"
  />
  <h3
    className="recipeText"
  >
    testing
  </h3>
</div>
`;
